doc/api/diagnostics_channel.md

```diff
@@ -0,0 +1,227 @@
+
+* `message` {any} The message to send to the channel subscribers
+
+Publish a message to any subscribers to the channel. This will trigger
+message handlers synchronously so they will execute within the same context.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.publish({
+  some: 'message'
+});
+```
+
+#### `channel.subscribe(onMessage)`
+
+* `onMessage` {Function} The handler to receive channel messages
+  * `message` {any} The message data
+  * `name` {string} The name of the channel
+
+Register a message handler to subscribe to this channel. This message handler
+will be run synchronously whenever a message is published to the channel. Any
+errors thrown in the message handler will trigger an [`'uncaughtException'`][].
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.subscribe((message, name) => {
+  // Received data
+});
+```
+
+#### `channel.unsubscribe(onMessage)`
+
+* `onMessage` {Function} The previous subscribed handler to remove
+
+Remove a message handler previously registered to this channel with
+[`channel.subscribe(onMessage)`][].
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.channel('my-channel');
+
+function onMessage(message, name) {
+  // Received data
+}
+
+channel.subscribe(onMessage);
+
+channel.unsubscribe(onMessage);
+```
+
+#### `channel.span(data)`
+
+* `data` {any} The message to send as the start event of the span
+* Returns: {Span} The span object
+
+Create a span object to tie a sequence of events together. This wraps the input
+data in a new object which includes an action type of `start`, `annotation`, or
+`end` and a numeric id to link span messages together for later reconsitution.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.channel('my-channel');
+
+const span = channel.span({
+  some: 'message'
+});
+```
+
+### Class: `Span`
+
+The class `Span` represents a series of messages to send to a named channel.
+It is used to automatically apply a shared id to all messages sent through the
+span enabling later reconsitution of the message sequence.
+
+#### `span.annotate(data)`
+
+* `data` {any} The message to send as an annotation event of the span
+
+Send an `annotation` event to the message sequence.
+
+This is useful for augmenting the span with additional data before completion.
+For example, a library may use a connection pool internally and want to report
+the host name of the acquired connection to a span representing a higher-level
+interaction.
+
+```js
+const diagnostics_channel = require('diagnostics_channel');
+
+const channel = diagnostics_channel.channel('my-channel');
+
+const span = channel.span({
+  some: 'message'
+});
+
+span.end({
+  some: 'other message'
+});
+```
```

Comment:
I think this example was supposed to be a `span.annotate(data)` example?

Reply:
Oops. Thanks. Fixed now. 😅